src/Symfony/Component/Translation/MessageSelector.php

```diff
@@ -47,11 +47,11 @@ class MessageSelector
      */
     public function choose($message, $number, $locale)
     {
-        $parts = explode('|', $message);
+        preg_match_all('/(?:\|\|.|[^\|])+/', $message, $parts);
```

Comment:
why ``.`` after the double pipe ? this would also match a pipe.

Btw, you should use a possessive quantifier to avoid useless backtracking